pkg/distributor/ha_tracker.go

```diff
@@ -80,27 +80,32 @@ type haTracker struct {
 // HATrackerConfig contains the configuration require to
 // create a HA Tracker.
 type HATrackerConfig struct {
+	EnableHATracker bool `yaml:"enable_ha_tracker,omitempty"`
 	// We should only update the timestamp if the difference
 	// between the stored timestamp and the time we received a sample at
 	// is more than this duration.
-	UpdateTimeout time.Duration
+	UpdateTimeout time.Duration `yaml:"ha_tracker_update_timeout"`
 	// We should only failover to accepting samples from a replica
 	// other than the replica written in the KVStore if the difference
 	// between the stored timestamp and the time we received a sample is
 	// more than this duration
-	FailoverTimeout time.Duration
+	FailoverTimeout time.Duration `yaml:"ha_tracker_failover_timeout"`
 
 	KVStore kv.Config
 }
 
-// RegisterFlags adds the flags required to config this to the given FlagSet
-func (cfg *HATrackerConfig) RegisterFlags(f *flag.FlagSet) {
+// RegisterFlags adds the flags required to config this to the given FlagSet. If prefix is not an empty string it should end with a period.
+func (cfg *HATrackerConfig) RegisterFlags(prefix string, f *flag.FlagSet) {
```

Comment:
TBH not sure the prefix is the right way to go here - its not like we're planning on using the HA tracker anywhere else, are we?

Reply:
Not that I can think of, but it makes the flag registration a bit cleaner and less chance of new flags missing the `distributor.` prefix imo. I can revert this if you disagree.

Reply:
We have to be really careful changing flag names.  For instance you dropped the dot between distributor and ha-tracker.  This has the potential to break rollouts and piss off users.

Reply:
https://github.com/cortexproject/cortex/pull/1546/commits/8ffe4b9064485e5cc293892209bdb391c8257e9a#r312024987